src/adapter/optimizations/item_lookup.rs

```diff
@@ -8,6 +8,7 @@ use trustfall::{
 };
 
 use super::super::{RustdocAdapter, origin::Origin, vertex::Vertex};
+use std::sync::Arc;
```

Comment:
nit: let's put this in the upper "external imports" block, not between `super` and `crate`